nilearn/_utils/param_validation.py

```diff
@@ -63,3 +68,77 @@ def check_threshold(threshold, data, percentile_func, name='threshold'):
         raise TypeError('%s should be either a number '
                         'or a string finishing with a percent sign' % (name, ))
     return threshold
+
+
+def _get_mask_volume(mask_img):
+    """Computes the volume of a brain mask in mm^3
+
+    Parameters
+    ----------
+    mask_img : nibabel image object
+        Input image whose voxel dimensions are to be computed.
+
+    Returns
+    -------
+    vol : float
+        The computed volume.
+    """
+    vox_dims = get_header(mask_img).get_zooms()[:3]
+    return 1. * np.prod(vox_dims) * mask_img.get_data().astype(np.bool).sum()
```

Comment:
You're assuming an orthonormal grid, aren't you. Why don't you take
np.abs(np.linalg.det (make_img.affine[:3, :3])) instead of np.prod(vox_dims) ?